src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/ExtendedQueryTagStoreEntry.cs

```diff
@@ -3,37 +3,45 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
+using EnsureThat;
+
 namespace Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag
 {
     /// <summary>
     /// Represent each extended query tag entry has retrieved from the store.
     /// </summary>
     public class ExtendedQueryTagStoreEntry : ExtendedQueryTagEntry
     {
-        public ExtendedQueryTagStoreEntry(int key, string path, string vr, string privateCreator, QueryTagLevel level, ExtendedQueryTagStatus status)
+        public ExtendedQueryTagStoreEntry(int key, string path, string vr, string privateCreator, QueryTagLevel level, ExtendedQueryTagStatus status, ExtendedQueryTagVersion? version)
```

Comment:
I am thinking we should just move the minimum version of `ExtendedQueryTag` to V4. If we do that, can we remove the complexity of determining whether this is a `null` value?

Reply:
I'm thinking what would happen to customer since we previously set the version to V2.
let's assume customer has database V2, when deployed with our latest binary, would their database be upgraded to V4 automatically?